### PR TITLE
Added serialize and deserialize functions in Matlab mym

### PR DESCRIPTION
**Pull request by saumilpatel**
Distributions created for Windows, Mac and Linux OSes. Only 64 bit OSes are supported.
Distributions tested in Matlab 2018a.
mym folder has testscript folder in which there is a script that describes how serialize and deserialize can be used.

### CODE REVIEW

File: src/mym.cpp [hunk 1]
```diff
@@ -888,6 +894,120 @@ void mexFunction(int nlhs, mxArray*plhs[], int nrhs, const mxArray*prhs[]) {
...
+            char** po = 0;                              // pointer to placeholders openning symbols
+            char** pc = 0;                              // pointer to placeholders closing symbols
+            bool*  pec = 0;                             // pointer to 'enable compression field'
+            char** pa = 0;                              // pointer to placeholder in-string argument
+            unsigned* ps = 0;                           // pointer to placeholders size (in bytes)
+            pfserial* pf = 0;                           // pointer to serialization function
+            // LOOK FOR THE PLACEHOLDERS
+            po = (char**)mxCalloc(nex+1, sizeof(char*));
+            pc = (char**)mxCalloc(nex+1, sizeof(char*));
+            if ((po[nac++] = strstr(query, PH_OPEN)))
```
Comment (dimitri-yatsenko) [inline src/mym.cpp]: Let's take the assignment out of the the if condition for better readability.

File: src/mym.cpp [hunk 2]
```diff
@@ -888,6 +894,120 @@ void mexFunction(int nlhs, mxArray*plhs[], int nrhs, const mxArray*prhs[]) {
...
+                    if (cmp_rate>MIN_CMP_RATE_ZLIB) {
+                        const size_t len_old = plen[i];
+                        plen[i] = LEN_ZLIB_ID+1+sizeof(_uint64)+len;
+                        pd[i] = (char*)mxRealloc(pd[i], plen[i]);
+                        memcpy(pd[i], (const char*)ZLIB_ID, LEN_ZLIB_ID);
+                        *(pd[i]+LEN_ZLIB_ID) = 0;
+
+                        *((_uint64*)(pd[i]+LEN_ZLIB_ID+1)) = (_uint64) len_old;
+                        memcpy(pd[i]+LEN_ZLIB_ID+1+sizeof(_uint64), pcmp, len);
+                        //BUG: *((_uint64*)(pd[i]+LEN_ZLIB_ID+1+sizeof(_uint64))) = (_uint64) len;
```
Comment (dimitri-yatsenko) [inline src/mym.cpp]: what is this for?

File: src/mym.cpp [hunk 3]
```diff
@@ -459,14 +465,14 @@ void mexFunction(int nlhs, mxArray*plhs[], int nrhs, const mxArray*prhs[]) {
         //mysql_options(conn, MYSQL_OPT_RECONNECT, &my_true);
         if (!mysql_real_connect(conn, host, user, pass, NULL, port, NULL, CLIENT_MULTI_STATEMENTS))
             mexErrMsgTxt(mysql_error(conn));
-        const char*c = mysql_stat(conn);
-        if (c) {
+        const char*cs = mysql_stat(conn); // return connection status as a string - changed variable from c to cs
```
Comment (dimitri-yatsenko) [inline src/mym.cpp]: This comment seems to be about the change rather than about the final status of the code.

File: src/mym.cpp (hunk 1 above)
Comment (saumilpatel) [reply src/mym.cpp]: Dimitri, this is copy from the original code, I am not sure if it is worthwhile to spend the time to just clean it in the section we have added

File: src/mym.cpp (hunk 2 above)
Comment (saumilpatel) [reply src/mym.cpp]: Again, it is copied from original code, I have now forgotten what this was for, you can see that the code is extremely difficult to follow

File: src/mym.cpp (hunk 3 above)
Comment (saumilpatel) [reply src/mym.cpp]: Just a comment to remind that mysql_stat(conn) returns connection status, done when tracing the code, got sick of single letter variable names :-)

**Comment by saumilpatel** [comment]
Have removed the .m~ file

**Comment by dimitri-yatsenko** [comment]
These changes were reworked and integrated in #37

**Pull request closed without merging**